src/Models/ModelInstantiator.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace PDGA\DataObjects\Models;
+
+use PDGA\DataObjects\Attributes\Column;
+
+use ReflectionProperty;
+use ReflectionException;
+
+class ModelInstantiator
+{
+    /**
+     * Converts an associative array to a validated instance of a data object class.
+     *
+     * @param array  $arr   an array of values corresponding to the public properties of the data object.
+     * @param string $class Fully-qualified class name of the data object.
+     *
+     * @throws ReflectionException
+     * @return object
+     */
+    public function arrayToDataObject(
+        array  $arr,
+        string $class
+    ): object
+    {
+        $instance = new $class();
+
+        // TODO validate.
+
+        // Assign all public properties.
+        foreach ($this->dataObjectPropertyColumns($class) as $property => $column)
+        {
+            $instance->{$property} = $arr[$property] ?? null;
+        }
+
+        return $instance;
+    }
+
+    /**
+     * Converts a data object to an associative array with keys matching database fields for the
+     * corresponding database model.
+     *
+     * @param object $data_object An instance of a hydrated data object.
+     *
+     * @throws ReflectionException
+     * @return array
+     */
+    public function dataObjectToDatabaseModel(
+        object $data_object
+    ): array
+    {
+        $model_array = [];
+
+        // Loop through all assigned properties of the object.
+        foreach ($this->dataObjectPropertyColumns($data_object::class) as $property => $column)
+        {
+            // Assign key based on the name property of the Column attribute.
+            $model_array[$column] = $data_object->{$property};
```

Comment:
Similar comment here as the one above: This needs to check if `property` is set, hopefully using the helper methods that Drew wrote in the `ValidationEnforcer`.  If `$property` is uninitialized, this will result in a warning or error depending on PHP config.

Reply:
Thanks. I added a basic check for the time being, and I'll plan to update once the validation code is merged.